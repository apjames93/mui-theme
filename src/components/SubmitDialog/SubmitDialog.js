import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import PropTypes from 'prop-types';
import copy from 'copy-to-clipboard';

const SubmitDialog = ({ theme, open, handleClose }) => {
  const copyTheme = () => {
    copy(JSON.stringify(theme));
  };

  return (
    <Dialog onClose={handleClose} aria-labelledby="simple-dialog-title" open={open}>
      <DialogTitle id="simple-dialog-title">Set backup account</DialogTitle>
      <DialogContent>
        {JSON.stringify(theme)}
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="secondary">
          close
        </Button>
        <Button onClick={copyTheme}>
          copy
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default SubmitDialog;

SubmitDialog.propTypes = {
  theme: PropTypes.shape(),
  open: PropTypes.bool,
  handleClose: PropTypes.func.isRequired,
};

SubmitDialog.defaultProps = {
  theme: {},
  open: false,
};
