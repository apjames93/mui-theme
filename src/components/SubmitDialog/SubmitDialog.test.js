import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import SubmitDialog from './SubmitDialog';

function setup() {
  const props = {};
  const comp = mount(<SubmitDialog {...props} />);
  return { comp, props };
}

describe('<SubmitDialog />', () => {
  it('renders SubmitDialog', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((<SubmitDialog {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
