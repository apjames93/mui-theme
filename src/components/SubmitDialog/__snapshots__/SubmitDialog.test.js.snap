// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitDialog /> snapshot 1`] = `null`;
